src/utils/loadAndCheckOptions.m

```diff
@@ -13,25 +13,29 @@
     opt = spm_select('FPList', pwd, '^options_task-.*.json$');
   end
 
-  if isstruct(opt)
-    opt = checkOptions(opt);
-    fprintf(1, '\nOptions are locked & loaded.\n\n');
-    return
-  end
-
   % finds most recent option file
   if size(opt, 1) > 1
     containsDate = cellfun(@any, strfind(cellstr(opt), '_date-'));
-    opt = opt(containsDate, :);
-    opt = sortrows(opt);
-    opt = opt(end, :);
+    if any(containsDate)
+      opt = opt(containsDate, :);
+      opt = sortrows(opt);
+      opt = opt(end, :);
+    end
+  end
+
+  if ischar(opt) && size(opt, 1) == 1
+    if exist(opt, 'file')
+      fprintf(1, '\nReading option from: %s.\n', opt);
+      opt = spm_jsonread(opt);
+    else
+      error('the requested file does not exist: %s', opt);
+    end
   end
 
-  if ischar(opt) && exist(opt, 'file')
-    fprintf(1, '\nReading option from: %s.\n', opt);
-    opt = spm_jsonread(opt);
-  else
-    error('the requested file does not exist: %s', opt);
+  % temporary hack to fix the way spm_jsonread reads some empty fields
+  % REPORT IT TO SPM
+  if isfield(opt, 'subjects') && ~iscell(opt.subjects) && isnan(opt.subjects)
+    opt.subjects = {[]};
```

Comment:
it's crashing atm. It cannot find `opt`. I think because of the previous if loop, `ischar(optionJsonFile)  == 0`

```
  if ischar(optionJsonFile) && size(optionJsonFile, 1) == 1
    if exist(optionJsonFile, 'file')
      fprintf(1, '\nReading option from: %s.\n', optionJsonFile);
      opt = spm_jsonread(optionJsonFile);
    else
      error('the requested file does not exist: %s', optionJsonFile);
    end
  end
```

Reply:
can you show me the json file you pass to the function, or do you pass it an `opt` structure directly?